app/lib/publisher.rb

```diff
@@ -0,0 +1,64 @@
+require "services"
+
+class Publisher
+  def self.publish_all(types: nil)
+    new(types).publish_all
+  end
+
+  def publish_all
+    types.each do |document_type|
+      publish_document_type(document_type)
+    end
+  end
+
+private
+
+  attr_reader :types
+
+  def initialize(types)
+    @types = types.presence || document_types
+  end
+
+  def publish_document_type(document_type)
+    content_id_and_locale_pairs_for_document_type(
+      document_type,
+    ).each do |content_id, locale|
+      PublishWorker.perform_async(content_id, locale)
```

Comment:
I reckon we can get away without doing this asynchronously. Publishing documents via the Publishing API is usually very quick and doing it synchronously will give us a better idea of progress (i.e. when the Rake tasks finishes, the requests have finished) plus it'll crash dramatically if an error does occur and give us better visibility.

Reply:
Your call, really :) I have no preference no history of why other similar features are done in asynch. Will get rid of the job :)